tests/kernel/kernel_constructors.cpp

```diff
@@ -21,276 +21,219 @@
 
 #include "../common/common.h"
 
-#define TEST_NAME kernel_constructors
-
 template <int a>
 class test_kernel {
  public:
   void operator()() const {}
 };
 
-class kernel0;
-class kernel1;
-
 namespace kernel_constructors__ {
 using namespace sycl_cts;
 
-/** test sycl::kernel
- */
-class TEST_NAME : public sycl_cts::util::test_base {
- public:
-  /** return information about this test
-   */
-  void get_info(test_base::info &out) const override {
-    set_test_info(out, TOSTRING(TEST_NAME), TEST_FILE);
-  }
+TEST_CASE("Test copy constructor", "[kernel]") {
+  cts_selector ctsSelector;
+  auto ctsQueue = util::get_cts_object::queue(ctsSelector);
+  auto deviceList = ctsQueue.get_context().get_devices();
+  auto ctx = ctsQueue.get_context();
+
+  using k_name = test_kernel<0>;
+  auto kb =
+      sycl::get_kernel_bundle<k_name, sycl::bundle_state::executable>(ctx);
+  auto kernelA = kb.get_kernel(sycl::get_kernel_id<k_name>());
 
-  /** execute the test
-   */
-  void run(util::logger &log) override {
-    {
-      /* Test copy constructor
-       */
-      {
-        cts_selector ctsSelector;
-        auto ctsQueue = util::get_cts_object::queue(ctsSelector);
-        auto deviceList = ctsQueue.get_context().get_devices();
-        auto ctx = ctsQueue.get_context();
-
-        using k_name = test_kernel<0>;
-        auto kb = sycl::get_kernel_bundle<
-                    k_name, sycl::bundle_state::executable>(ctx);
-        auto kernelA = kb.get_kernel(sycl::get_kernel_id<k_name>());
-
-        sycl::kernel kernelB(kernelA);
-
-        ctsQueue.submit([&](sycl::handler &cgh) {
-          cgh.single_task(k_name());
-        });
+  sycl::kernel kernelB(kernelA);
+
+  ctsQueue.submit([&](sycl::handler &cgh) { cgh.single_task(k_name()); });
 
 #ifdef SYCL_BACKEND_OPENCL
-        if (ctsQueue.get_backend() == sycl::backend::opencl) {
-          auto iopKernelA =
-              sycl::get_native<sycl::backend::opencl>(kernelA);
-          auto iopKernelB =
-              sycl::get_native<sycl::backend::opencl>(kernelB);
-          if (!ctsSelector.is_host() && (iopKernelA != iopKernelB)) {
-            FAIL(log,
-                 "kernel was not constructed correctly. (contains different "
-                 "OpenCL kernel object)");
-          }
-        }
+  if (ctsQueue.get_backend() == sycl::backend::opencl) {
+    auto iopKernelA = sycl::get_native<sycl::backend::opencl>(kernelA);
+    auto iopKernelB = sycl::get_native<sycl::backend::opencl>(kernelB);
+    if (!ctsSelector.is_host() && (iopKernelA != iopKernelB)) {
+      FAIL(
+          "kernel was not constructed correctly. (contains different "
+          "OpenCL kernel object)");
+    }
```

Comment:
Should we use `CHECK` instead of `if` + `FAIL`?

Reply:
Changed it.

Reply:
Thanks!